crates/iota-indexer/tests/ingestion_tests.rs

```diff
@@ -318,15 +331,20 @@ mod ingestion_tests {
         indexer_wait_for_checkpoint(&pg_store, 1).await;
 
         // Read the transaction from the database directly.
-        let actual_insertion_order = read_only_blocking!(&pg_store.blocking_cp(), |conn| {
-            tx_insertion_order::table
-                .filter(tx_insertion_order::tx_digest.eq(digest.inner().to_vec()))
-                .select(tx_insertion_order::insertion_order)
-                .first::<i64>(conn)
+        let stored = read_only_blocking!(&pg_store.blocking_cp(), |conn| {
+            tx_global_order::table
+                .filter(tx_global_order::tx_digest.eq(digest.inner().to_vec()))
+                .select(TxGlobalOrder::as_select())
+                .first::<TxGlobalOrder>(conn)
         })
-        .context("Failed reading tx insertion order from PostgresDB")?;
+        .context("Failed reading `tx_global_order` from PostgresDB")?;
 
-        assert_eq!(actual_insertion_order, pre_existing_insertion_order);
+        assert_eq!(stored.global_sequence_number, global_sequence_number);
+        let expected_optimistic_sequence_number = 1;
```

Comment:
The sequence number for optimistic transactions will be `SELECT MAX(tx_sequence_number) FROM tx_digests` at the time of insertion. So in that case it is 1. What I can't figure from the code changes, where does the optimistic sequence number field get set? Between
https://github.com/iotaledger/iota/pull/7152/files#diff-24c4275b8eee2f1692509dbc74e4f8aab2a378b8e83093f1b2237d47e5984f29R308-R313 we only populate the `tx_global_order` table with `optimistic_sequence_number: None`?

Reply:
This patch handles the checkpoint transactions only. See the `committer.rs` changes.

For optimistic transactions we will just set `optimistic_sequence_number` to `None` and rely on the postgres sequence to generate it on the db server side.